hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -394,6 +394,12 @@ private boolean shouldCompareWithFile(BloomIndexFileInfo indexInfo, String recor
       if (v1._2().isPresent()) {
         String filename = v1._2().get();
         if (filename != null && !filename.isEmpty()) {
+          // If a record being tagged exists in multiple files in the same partition and if the
+          // currentLocation has already been set, then create a new record with the same key
+          // and data, set the current path of the new record and return it.
+          if (record.getCurrentLocation() != null) {
```

Comment:
whats the intent of this if-block? Seems you are re-initing the same object again based on it self?

Reply:
When you a record in multiple files in the same partition, then rowKeyRecordPairRDD will have 2 entries with the same exact in memory copy of the HoodieRecord and the 2 separate filenames that the record is found in. In this case, setCurrentLocation will fail for the second file, as we are setting the location on the same in memory object. HoodieRecord restricts setting the currentLocation if it is already set. So creating multiple copies of the HoodieRecord with the different file locations. 

I can create a new HoodieRecord here and set the current location. Thoughts ?

Reply:
I see. this is very hacky :) .. arghhh. :)
can you explain this more clearly in the comments there? esp this part "When you a record in multiple files in the same partition, then rowKeyRecordPairRDD will have 2 entries with the same exact in memory copy of the HoodieRecord and the 2 separate filenames that the record is found in" 

we can leave as it is.